cli/azd/pkg/azure/resource_ids.go

```diff
@@ -6,10 +6,24 @@ package azure
 import (
 	"fmt"
 	"regexp"
+	"strings"
 
 	"github.com/azure/azure-dev/cli/azd/pkg/convert"
 )
 
+// SubscriptionFromRID returns the subscription id component of a resource, or empty if the ID does not
+// contain a subscriptions component.
+func SubscriptionFromRID(rid string) string {
+	parts := strings.Split(rid, "/")
+	for idx, part := range parts {
+		if part == "subscriptions" && idx+1 < len(parts) {
+			return parts[idx+1]
+		}
+	}
+
+	return ""
```

Comment:
We call` len(parts)` at most once.  If the segment we are looking at is "subscriptions" we'll look at it. If for some reason the test failed, it would mean we were at the end of the range of parts, and so there would be no more iterations of the loop (and hence no more calls to len).